jujugui/static/gui/src/app/store/env/api.js

```diff
@@ -222,6 +222,14 @@ YUI.add('juju-env-api', function(Y) {
       this.pendingLoginResponse = false;
       this.on('_rpc_response', this._handleRpcResponse);
       this._allWatcherBeingStopped = false;
+      // When using GISF and switching between models or going from the
+      // unconnected state to a model the env properties are reset, but only
+      // those that were previously set. This meant that the environmentName
+      // would not be updated as when the model disconnected then the value was
+      // not reset. The reason we need the value to be reset is so that when it
+      // is set to the new value it fires a value change event, which in turn
+      // updates the name set in the db. See onEnvironmentNameChange.
+      this.setConnectedAttr('environmentName', undefined);
```

Comment:
If the name is `null` then the model name gets displayed as such instead of displaying 'untitled-model'.

As for why it's done there, we only need to set the name when using the juju api, when using the controller api it works fine so we moved this set to the api module.